lib/session.go

```diff
@@ -6,26 +6,39 @@
 package lib
 
 import (
+	"context"
 	"os"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/credentials"
 )
 
-func newAWSConfig(cfg *Config) (aws.Config, error) {
-	config := aws.Config{
-		Region:      cfg.RegionName,
-		Credentials: createCredentials(cfg),
+func newAWSConfig(ctx context.Context, cfg *Config) (aws.Config, error) {
+	// Build options for LoadDefaultConfig
+	var opts []func(*config.LoadOptions) error
+
+	if cfg.EndpointURL != "" {
+		opts = append(opts, config.WithEndpointResolverWithOptions(
+			aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
```

Comment:
There are compile/test failures, but if you run `staticcheck` (part of the CI build) you will also see that this construct is deprecated. This is now handled in the newRemoteStore func.

Reply:
I think this is a case of me not understanding the go ecosystem, both `go test` and `go build` are passing. Are there other commands I should be using?